### PR TITLE
fifo: log non-ENOENT unlink() errors

### PR DESCRIPTION
**Pull request by awelzel**
Maybe this could help debugging #1453 .

### CODE REVIEW

File: core/fifo.c [hunk 1]
```diff
@@ -104,7 +104,12 @@ int uwsgi_master_fifo() {
 
 	char *path = uwsgi_fifo_by_slot();
 
-	unlink(path);
+	if (unlink(path)) {
+		if (errno != ENOENT) {
+			uwsgi_error("uwsgi_master_fifo()/unlink()");
+			// exit(1); // XXX: If unlink() failed, mkfifo() will probably as well
```
Comment (xrmx) [inline core/fifo.c]: Either exit() or don't, both are fine by me,  but let's not add commented out code please :)

Comment (xrmx) [reply core/fifo.c]: To be on the safe side better to don't exit though

Comment (awelzel) [reply core/fifo.c]: Sounds sensible - thanks. Also changed to `&& errno != ENOENT`.

**Comment by xrmx** [comment]
Thanks!

**Pull request merged**